src/rhub/api/bare_metal/provision.py

```diff
@@ -0,0 +1,106 @@
+import logging
+from io import BytesIO
+
+from connexion import problem
+from flask import send_file
+
+from rhub.api import db
+from rhub.bare_metal.model import (
+    BareMetalHost,
+    BareMetalHostStatus,
+    BareMetalImage,
+    BareMetalImageISO,
+    BareMetalImageQCOW2,
+    BareMetalProvision,
+    BareMetalProvisionISO,
+    BareMetalProvisionQCOW2,
+    bare_metal_provision_full,
+)
+from rhub.bare_metal.tasks import ironic_provision_stop_task, ironic_provision_task
+
+logger = logging.getLogger(__name__)
+
+
+def provision_list():
+    provisions = db.session.query(bare_metal_provision_full)
+    return {
+        "data": [i.to_dict_with_super() for i in provisions.all()],
+        "total": provisions.count(),
+    }
+
+
+def provision_create(body):
```

Comment:
There is nothing here to tell user what you expect in the body. Also what exactly does 'provision' plan to achieve?  Some description details in comments at least for the file as a whole in what it's purpose in life is.  The same for other methods doing create.

Reply:
Once we agree on the model (this PR), I'll update and add the documentation similar to: https://github.com/resource-hub-dev/rhub-api/blob/master/src/rhub/openapi/tower.yml